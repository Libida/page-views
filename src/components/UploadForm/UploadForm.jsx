import React from 'react';

import './UploadForm.scss';

const UploadForm = ({
                      acceptTypes = '.txt',
                      onChange,
                      onReset,
                      shouldShowReset = false,
                      uploadRef,
                    }) => {
  return (
    <section className="upload-form-section">
      <h2 className="h-2">Add a Document</h2>
      <form className="upload-form">
        <input
          type="file"
          className="upload-form__choose-file"
          accept={acceptTypes}
          onChange={onChange}
          ref={uploadRef}
        />
        {shouldShowReset && (
          <button
            type="button"
            className="upload-form__reset"
            aria-label="Reset"
            title="Reset"
            onClick={onReset}
          />
        )}
      </form>
    </section>
  );
};

export default UploadForm;
