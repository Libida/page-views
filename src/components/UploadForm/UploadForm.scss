@import './src/styles/variables';

.upload-form {
  margin-top: 20px;
  display: flex;
  align-items: center;
}

.upload-form__reset {
  border: 0;
  padding: 0;
  background-color: transparent;

  &:after {
    display: inline-block;
    content: '\00d7';
    font-size: 1.2rem;
    color: $red-1;
  }
}

.upload-form__choose-file, .upload-form__reset {
  cursor: pointer;
}
