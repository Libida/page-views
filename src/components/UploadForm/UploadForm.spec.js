import React from 'react';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { shallow, configure } from 'enzyme';

configure({adapter: new Adapter()});

import UploadForm from './UploadForm';

describe('UploadForm', () => {
  const initialProps = {
    onChange: () => {},
    onReset: () => {},
    uploadRef: [],
  };
  const wrapper = (props) => shallow(<UploadForm {...props} />);

  test('should match snapshot for default initial props', () => {
    expect(wrapper(initialProps)).toMatchSnapshot();
  });

  test('should match snapshot for case with visible reset button', () => {
    expect(wrapper({...initialProps, shouldShowReset: true})).toMatchSnapshot();
  });
});
