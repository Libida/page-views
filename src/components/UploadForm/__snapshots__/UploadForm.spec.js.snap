// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadForm should match snapshot for case with visible reset button 1`] = `
<section
  className="upload-form-section"
>
  <h2
    className="h-2"
  >
    Add a Document
  </h2>
  <form
    className="upload-form"
  >
    <input
      accept=".txt"
      className="upload-form__choose-file"
      onChange={[Function]}
      type="file"
    />
    <button
      aria-label="Reset"
      className="upload-form__reset"
      onClick={[Function]}
      title="Reset"
      type="button"
    />
  </form>
</section>
`;

exports[`UploadForm should match snapshot for default initial props 1`] = `
<section
  className="upload-form-section"
>
  <h2
    className="h-2"
  >
    Add a Document
  </h2>
  <form
    className="upload-form"
  >
    <input
      accept=".txt"
      className="upload-form__choose-file"
      onChange={[Function]}
      type="file"
    />
  </form>
</section>
`;
